### ISSUE TITLE
question: Class Validator throws ',' expected when I run "tsc"

### ISSUE THREAD

**Issue by SkullCutter02** [typestack/class-validator#881]
### Description

I'm using TypeORM and Class Validator together. I need to deploy my app to heroku now, so I created a script called "npm build", which runs the command "tsc". However, it shows these errors which made my app unable to deploy. Below are the errors:

**Minimal code-snippet showcasing the problem**
 

```ts
node_modules/class-validator/types/decorator/common/IsIn.d.ts:6:71 - error TS1005: ',' expected.

6 export declare function isIn(value: unknown, possibleValues: readonly unknown[]): boolean;
                                                                        ~~~~~~~

node_modules/class-validator/types/decorator/common/IsIn.d.ts:6:78 - error TS1005: ',' expected.

6 export declare function isIn(value: unknown, possibleValues: readonly unknown[]): boolean;
                                                                               ~

node_modules/class-validator/types/decorator/common/IsIn.d.ts:10:47 - error TS1005: ',' expected.

10 export declare function IsIn(values: readonly any[], validationOptions?: ValidationOptions): PropertyDecorator;
                                                 ~~~

node_modules/class-validator/types/decorator/common/IsIn.d.ts:10:50 - error TS1005: ',' expected.

10 export declare function IsIn(values: readonly any[], validationOptions?: ValidationOptions): PropertyDecorator;
                                                    ~

node_modules/class-validator/types/decorator/common/IsNotIn.d.ts:6:74 - error TS1005: ',' expected.

6 export declare function isNotIn(value: unknown, possibleValues: readonly unknown[]): boolean;
                                                                           ~~~~~~~

node_modules/class-validator/types/decorator/common/IsNotIn.d.ts:6:81 - error TS1005: ',' expected.

6 export declare function isNotIn(value: unknown, possibleValues: readonly unknown[]): boolean;
                                                                                  ~

node_modules/class-validator/types/decorator/common/IsNotIn.d.ts:10:50 - error TS1005: ',' expected.

10 export declare function IsNotIn(values: readonly any[], validationOptions?: ValidationOptions): PropertyDecorator;
                                                    ~~~

node_modules/class-validator/types/decorator/common/IsNotIn.d.ts:10:53 - error TS1005: ',' expected.

10 export declare function IsNotIn(values: readonly any[], validationOptions?: ValidationOptions): PropertyDecorator;
                                                       ~


Found 8 errors.

```

### Expected behavior

I've tried uninstalling and reinstalling class validator, but in vain. I expect it to have zero errors and my app successfully deployed to heroku. 

### Actual behavior

What happens is that the errors above are thrown, making me unable to deploy my application.

**Comment by NoNameProvided** [comment]
You should not compile your `node_modules`. Add it to the exclude list in your `tsconfig.json`.

**Comment by SkullCutter02** [comment]
> You should not compile your `node_modules`. Add it to the exclude list in your `tsconfig.json`.

Just added "exclude": ["node_modules"] to tsconfig.json, still same problem.

**Comment by SkullCutter02** [comment]
I'd also like to add that if I just run ```tsc``` in my terminal, these errors will not log. However, if I did ```npm run build``` which executes the command ```tsc```, then it will throw these errors. Very weird

**Comment by NoNameProvided** [comment]
I am almost 100% sure this is some problem with your build setup. Please provide a **minimal, reproducible** example otherwise I can guess only.

**Comment by YohannesDK** [comment]
I have the exact same problem...

Here is the log the run,
[2021-01-24T01_22_25_002Z-debug.log](https://github.com/typestack/class-validator/files/5861422/2021-01-24T01_22_25_002Z-debug.log)

My package.json
![image](https://user-images.githubusercontent.com/54357153/105618966-0278c180-5dee-11eb-9d4c-12ceda487623.png)

My tsconfig.json
![image](https://user-images.githubusercontent.com/54357153/105618969-14f2fb00-5dee-11eb-9244-ef58798d4d24.png)

The error i get:
![image](https://user-images.githubusercontent.com/54357153/105618982-35bb5080-5dee-11eb-9622-3233e2cf063d.png)

And so on...

**Comment by YohannesDK** [comment]
Hi I had the same issue. I didn't manage to fix it completely, but when I ran `tsc -w` it compiled just fine. You could maybe try that, instead of running npm build?

**Comment by SkullCutter02** [comment]
K sorry for the late reply. I got a reproducible example that can be set up easily. 

First do ```npx typeorm init``` to initialise the project. Update the dependency version with ```npm install```, then install class-validator by doing: ```npm i class-validator```. When the project is initialised, it should have created an entity called User. Add a @MinLength decorator to one of the columns in the user entity. 

Afterwards, in package.json, create a script called build: ```"build": "tsc"```. Now run ```npm run build```, and the errors will be produced.

**Comment by vlapo** [comment]
What typescript version do you use? I think you need version `^3.1.3`. Maybe try `tsc -v` command.

**Comment by SkullCutter02** [comment]
> What typescript version do you use? I think you need version `^3.1.3`. Maybe try `tsc -v` command.

It's version 4.1.3

**Comment by NoNameProvided** [comment]
I cannot reproduce. After following the provided steps the starter successfully builds for me. It's some problem with your setup.

**Comment by SuperITMan** [comment]
Dear @NoNameProvided 
I'm currently working on a framework based on Angular 7 with TypeScript 3.2.4 and I encountered the same issue 😕 

To understand what's is going on, I created the following project https://github.com/SuperITMan/class-validator-ts-issue with a basic example in ng7, ng8 and ng9. ng7 and ng8 do not compile by running `npm run tsc`, you can find all the information in the README file 😊 

Could you reopen this issue and investigate it ? I have the feeling it might be related to TypeScript version...

Thanks for your great library!
Regards,
Alexis

**Comment by mdolr** [comment]
Hello,
I'm encoutering the same issue as @SkullCutter02 using the typeorm init code
Typescript version : 3.3.3333
Class-validator: ^0.13.1

**Comment by mdolr** [comment]
Hello @SkullCutter02 @SuperITMan, I had the same issue, fixed it by using Typescript v4.0.5 and adding --skipLibCheck or settings skipLibCheck to true in tsconfig.json

Hope this can help you
